MantidPlot/src/ProjectRecovery.cpp

```diff
@@ -767,4 +717,102 @@ std::vector<Poco::Path> ProjectRecovery::getRecoveryFolderCheckpointsPR(
   return getRecoveryFolderCheckpoints(recoveryFolderPath);
 }
 
+std::vector<std::string>
+ProjectRecovery::findOlderCheckpoints(const std::string &recoverFolder,
+                                      const std::vector<int> &possiblePids) {
+  // Currently set to a month in microseconds
+  const int64_t timeToDeleteAfter = 2592000000000;
```

Comment:
`constexpr Poco::Timestamp::TimeDiff timeToDeleteAfter = 1000000 * 60 * 60 * 24 * 30` would be clearer.

Reply:
:+1:

Reply:
Ah, I am unable to compile it as it overflows after the 100000 * 60 * 60, I believe it has to do with how it calculates the expression in the compiler.

Reply:
Ah yes, you'd have to specify the length otherwise it will assume the default integer width. `constexpr Poco::Timestamp::TimeDiff timeToDeleteAfter(1000000L * 60 * 60 * 24 * 30);` works

Reply:
:+1:

Reply:
Even if I turn each one into a long type it throws a warning of overflow on Windows, so I am going to change it back. :(